trader/trader.go

```diff
@@ -107,6 +109,7 @@ func MakeTrader(
 		dataKey:                        dataKey,
 		alert:                          alert,
 		metricsTracker:                 metricsTracker,
+		start:                          start,
```

Comment:
In general I prefer calling variables like “startTime” instead of a more generic “start” to avoid having to refer to the type all the time for vars on structs, since they are not defined close to usage. For short scoped vars in functions it makes sense for shorter names like “start”.

Reply:
and as a convention input parameters on factory methods should have the same name as the field, overriding the short scope naming convention (camel cased if strict field is exported)